### PR TITLE
Audio-visual speech recognition using GRID/CHiME data

### PR DESCRIPTION
**Pull request by hmeutzner**

### CODE REVIEW

**Comment by danpovey** [comment]
@jtrmal will hopefully have a look at this today or to-morrow.

**Comment by vijayaditya** [comment]
I can review this, if getting it done on Sunday is fine .

**Comment by danpovey** [comment]
Thanks-- we'll see if Yenda gets to it by then, if not that would be great.

File: egs/grid/s5/path.sh [hunk 1]
```diff
@@ -0,0 +1,15 @@
+# Default working directory for storing data/ exp/ mfcc/ etc
```
Comment (jtrmal) [inline egs/grid/s5/path.sh]: please use the default path.sh (KALDI_ROOT should be set differently and PATH as well)

as for the REC_ROOT, WAV_ROOT -- I'd suggest putting them into the main script, if viable.

File: egs/grid/README.md [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+H. Meutzner, N. Ma, R. Nickel, C. Schymura, D. Kolossa, "Improving Audio-Visual Speech Recognition using Deep Neural Networks with Dynamic Stream Reliability Estimates", ICASSP, New Orleans, USA, March 2017.
+
+Future releases will also contain the late-integration approach using dynamic stream weights.
+
+
+Data description
+--------------------------
+
+The experiments are based on the audio data of the CHiME-2 challenge [1] and the video data of the GRID audio-visual speech corpus [2,3].
+
```
Comment (jtrmal) [inline egs/grid/README.md]: Please describe that the data set is not exactly the same as for Chime as you remove one (?) speaker, i.e. the numbers are not directly comparable with the egs/chime2 recipe numbers.

File: egs/grid/s5/cmd.sh [hunk 3]
```diff
@@ -22,8 +22,18 @@ export mkgraph_cmd="queue.pl -l arch=*64,ram_free=4G,mem_free=4G"
...
+#c) USFD cluster options
+config="conf/queue_usfd.conf"
+#export train_cmd="queue.pl  --config $config --mem 4G --rmem 4G"
+#export decode_cmd="queue.pl  --config $config --mem 4G --rmem 4G"
+#export mkgraph_cmd="queue.pl  --config $config --mem 4G --rmem 4G"
+#export cuda_cmd="queue.pl  --config $config --mem 24G --rmem 20G --gpu 1 --time 24:00:00"
+
+
+#d) run it locally...
+export train_cmd=run.pl
```
Comment (jtrmal) [inline egs/grid/s5/cmd.sh]: I think the default config should be
export train_cmd="queue.pl --mem 2G"
export decode_cmd="queue.pl --mem 2G"
export mkgraph_cmd="queue.pl --mem 4G"
export cuda_cmd="queue.pl --gpu 1"

not the run.pl commands

File: egs/grid/s5/local/check_prerequisites.sh [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+else
+  if [ ! -d $VIDEO_ROOT ]; then
+    echo "Cannot find video directory $VIDEO_ROOT."
+    echo "Please unset video directory in path.sh or specify a valid location."
+    exit 1;
+  fi
+fi
+
+# Setup symlinks for steps and utils
+if [ ! -e "steps" ]; then
```
Comment (jtrmal) [inline egs/grid/s5/local/check_prerequisites.sh]: the steps and utils/ symlink should be already part of this PR (please add those), no need to do this.

File: egs/grid/s5/local/make_video.sh [hunk 5]
```diff
@@ -0,0 +1,112 @@
...
+# Apache 2.0.
+
+# This script prepares the video features such that they match the audio data of CHiME-1/2.
+
+# Begin configuration section.
+nj=4
+cmd=run.pl
+audioRoot="./proc/wav"
+videoRoot="./proc/video"
+
```
Comment (jtrmal) [inline egs/grid/s5/local/make_video.sh]: from linux/unix background, "proc" is confusing/misleading

File: egs/grid/s5/run_dnn.sh [hunk 6]
```diff
@@ -0,0 +1,333 @@
...
+
+# Setup other relevant directories
+data="$REC_ROOT/data"
+lang="$data/lang"
+dict="$data/local/dict"
+langtmp="$data/local/lang"
+mkdir -p $langtmp
+steps="steps"
+utils="utils"
+
```
Comment (jtrmal) [inline egs/grid/s5/run_dnn.sh]: I don't think the possibility to override the names of steps/ and utils/ directory is really useful, can you please remove it?

**Comment by hmeutzner** [comment]
Thanks for this fast review :-)

**Comment by danpovey** [comment]
@hmeutzner, what's happening with this?  I believe Yenda asked for changes; I don't see that you guys have made the changes.

**Comment by danpovey** [comment]
We haven't heard from you guys in a long time.  Should we close this PR?

**Pull request closed without merging**